### ISSUE TITLE
Allow bots to be more modular

### ISSUE THREAD

**Issue by meew0** [meew0/discordrb#48]
Currently, bots are not modular at all - you can achieve limited modularity in a hacky way by making methods that accept a `Bot` and add events to it. I'd rather have something like this:

```rb
module BotEvents
  include Discordrb::EventContainer
  
  presence(status: :online) do |event|
    puts "#{event.user} is online"
  end

  # ...
end

module BotCommands
  include Discordrb::CommandContainer
  
  command :hi { 'Hello!' }
  # ...
end

bot = Discordrb::Bot.new '...', '...'
bot.include BotEvents
bot.include BotCommands
bot.run
```

(Names of course subject to change)

- [x] Event containers
- [ ] Command containers

**Comment by meew0** [comment]
- [ ] Event containers
- [ ] Command containers

**Comment by meew0** [comment]
Event containers were implemented in 7a9a156f5652f67ee760f577d2ac5909da36c3d1

**Comment by meew0** [comment]
Command containers were implemented in https://github.com/meew0/discordrb/commit/29cc321ba961d6a7d1337a1698f6a9f42f076b4b
